src/App.tsx

```diff
@@ -4,64 +4,76 @@ import { ChannelEntry } from "./ChannelEntry";
 import { API_KEY } from "./BotAPI";
 import "./App.css";
 
-let channels: ChannelEntry[] = [];
+function Channels(): any {
+  const [channelsArray, updateChannelsArray] = useState<ChannelEntry[]>([]);
```

Comment:
keep a standard name for the useState hooks, so setChannelsArray or setChannels would be good :P